substrate/client/network/src/protocol/notifications/behaviour.rs

```diff
@@ -399,6 +399,14 @@ pub enum NotificationsOut {
 		/// Message that has been received.
 		message: BytesMut,
 	},
+
+	/// The remote peer has misbehaved and the connection has been closed.
```

Comment:
I would explain that right now this corresponds to the specific case of receiving data in outbound substream.